routing/datasource.go

```diff
@@ -520,11 +521,27 @@ func processRouteDefs(o Options, fr filters.Registry, defs []*eskip.Route) (rout
 
 type routeTable struct {
 	m             *matcher
+	once          sync.Once
+	routes        []*Route // only used for closing
 	validRoutes   []*eskip.Route
 	invalidRoutes []*eskip.Route
 	created       time.Time
 }
 
+// Close routeTable will cleanup all underlying resources, that could
+// leak goroutines.
+func (rt *routeTable) Close() {
```

Comment:
yes but it looks so ugly in case you have syntax highlighting, because `close` is an internal func like println. I agree will make it lower case, but I wait to finish the tests